### PR TITLE
Create new chart baseline with Chartjs4

### PR DESCRIPTION
**Pull request by Theeraphat-Sorasetsakul**
## Description
Rewrite ef-chart to use chart.js v4

Fixes # ([STG-197)](https://jira.refinitiv.com/browse/STG-197))

## Type of change

- [x] Refactor (improves code without changing its functionality)
- [x] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

This Pr create for baseline of ef-chart v7.  So, Tests, Demos, and other features are haven't included in this PR.

### CODE REVIEW

File: packages/elements/src/chart/__demo__/index.html [hunk 1]
```diff
@@ -14,7 +14,114 @@
     import '@refinitiv-ui/demo-block';
   </script>
 
-  <demo-block layout="normal" header="Line Chart - No Legend">
+  <demo-block layout="normal" header="Basic line">
+    <ef-chart id="line"></ef-chart>
+    <script>
+      const addTitle = () => {
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/__demo__/index.html]: unused `addTitle()`

File: packages/elements/src/chart/__demo__/index.html [hunk 2]
```diff
@@ -14,7 +14,114 @@
...
+        line.config.options.plugins.title.display = true;
+        line.config.options.plugins.title.text = 'Price of TRI.N in 2016';
+        line.updateChart();
+      };
+      const removeTitle = () => {
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/__demo__/index.html]: unused `removeTitle{}`

File: packages/elements/src/chart/index.ts [hunk 3]
```diff
@@ -13,62 +10,25 @@ import { customElement } from '@refinitiv-ui/core/decorators/custom-element.js';
...
-  DatasetColors
-} from './helpers/types';
 
-import type { Header } from '../header';
-import '../header/index.js';
-import '../layout/index.js';
+import {
+  Chart as ChartJS
+// eslint-disable-next-line import/extensions
+} from 'chart.js/auto';
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/index.ts:23]: We agreed to import only common chart tyles only, right? Let's add a todo reminding us to update this import later.

File: packages/elements/src/chart/index.ts [hunk 4]
```diff
@@ -13,62 +10,25 @@ import { customElement } from '@refinitiv-ui/core/decorators/custom-element.js';
...
-declare global {
-  interface Window {
-    Chart: ChartJS;
-  }
-}
+import type { Header } from '../header';
+import '../header/index.js';
 
-window.Chart.pluginService.register(doughnutCenterPlugin);
+export * from 'chart.js';
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/index.ts:31]: the same idea with importing `'chart.js/auto'` should apply here as well. Let's add a todo.

File: packages/elements/src/chart/index.ts [hunk 5]
```diff
@@ -447,24 +199,11 @@ export class Chart extends BasicElement {
...
 
-      // Create chart
-      this.chart = new window.Chart(ctx, this.config) as ChartJS;
-
-      if (isReusingCanvas) {
-        // If we're reusing an old chart canvas, we need to resize it.
-        // Destroying a chart has some strange side-effects on the canvas.
-        this.chart.resize();
-      }
+      this.chart = new ChartJS(this.canvas, this.config);
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/index.ts:210]: As we no longer destoy existing charts before creating a new one, I'm concerned that we might get memeory leak espcially during `config` prop update. Could you confirm this?

File: packages/elements/src/chart/index.ts [hunk 6]
```diff
@@ -113,40 +73,18 @@ export class Chart extends BasicElement {
...
-    }
-
-    return '';
-  }
-
-  /**
-   * Safely returns a dataset array
-   * @returns dataset array
-   */
-  protected get datasets (): Chart.ChartDataSets[] {
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/index.ts:146]: we don't need this API anymore?

**Review by wattachai-lseg — CHANGES_REQUESTED** [review]
Are you getting quite a number of errors from TypeScript e.g.  `this.config` error in `this.chart = new ChartJS(this.canvas, this.config);`

**Review by wattachai-lseg — CHANGES_REQUESTED** [review]
Are you getting quite a number of errors from TypeScript e.g.  `this.config` error in `this.chart = new ChartJS(this.canvas, this.config);`?

File: packages/elements/src/chart/index.ts (hunk 3 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/chart/index.ts:23]: Updated.

File: packages/elements/src/chart/index.ts (hunk 4 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/chart/index.ts:31]: Updated.

File: packages/elements/src/chart/index.ts (hunk 6 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/chart/index.ts:146]: I think it hasn't been used yet. So I deleted it first.

File: packages/elements/src/chart/index.ts (hunk 5 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/chart/index.ts:210]: After rechecked, it needs to destroy. So, add it above.

File: packages/elements/src/chart/index.ts [hunk 7]
```diff
@@ -447,24 +201,12 @@ export class Chart extends BasicElement {
...
    */
   protected createChart (): void {
-    const ctx = this.canvas.getContext('2d');
-    if (ctx && this.config) {
-
-      // Are we reusing an old chart canvas?
-      const isReusingCanvas = this.destroyChart();
-
-      // Preparing the resources before create chart
+    if (this.canvas && this.config) {
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/index.ts]: Let's check for canvas content as well:
```suggestion
    const ctx = this.canvas.getContext('2d');
      if (ctx && this.config) {
```

**Review by wattachai-lseg — APPROVED** [review]
As this is a part of a big upgrade, we shall allow some test issues and warnings. LGTM.

**Pull request merged**